CRM/Admin/Form/Options.php

```diff
@@ -158,7 +158,7 @@ public function buildQuickForm() {
     $this->add('text',
       'label',
       ts('Label'),
-      CRM_Core_DAO::getAttribute('CRM_Core_DAO_OptionValue', 'label'),
+      CRM_Core_DAO::getAttribute('CRM_Core_DAO_OptionValue', 'label'),// The label field which causes the issue. Something here or in getAttribute might need fixing.
```

Comment:
I strongly feel that something must be wrong here. However the first thing getAttribute does is to make a new DAO object which should connect using the "SET NAMES utf8" query.